emscripten.py

```diff
@@ -536,13 +536,16 @@ def finalize_wasm(infile, outfile, memfile):
 
   expected_exports = set(settings.EXPORTED_FUNCTIONS)
   expected_exports.update(asmjs_mangle(s) for s in settings.REQUIRED_EXPORTS)
+  for deps in js_syms.values():
+    expected_exports.update(asmjs_mangle(s) for s in deps)
```

Comment:
Its this map which is returned by `--symbols-only` which gives a map of JS symbols to their dependencies.   In my no recent change I added a comment here to explain what is going on

Reply:
I understand more now, but why is the assumption in the comment a safe one? What if one of them was explicitly marked for exporting?

Reply:
That could potentially be an issue yes.   We don't have a good way today to tell why a symbol was exported.

Note that he problem already exists even without this PR though for the `settings.REQUIRED_EXPORTS` list on the line above.  Admittedly that is not under user control so is less dangerous.

But the existing deps_info.py system works by adding entries to `settings.REQUIRED_EXPORTS` so is vulnerable to the exact same problem.

Ultimately we would like the output of the linker to somehow tell us explicitly which symbols were marked with EMSCRIPTEN_KEEPALIVE rather than trying to infer this from the export list.    But for now I think this approach is good enough for now, and at least should not be a regression.